client/data/home/use-skip-current-view-mutation.ts

```diff
@@ -31,14 +32,31 @@ function useSkipCurrentViewMutation( siteId: number ): Result {
 					path: `/sites/${ siteId }/home/layout/skip`,
 					apiNamespace: 'wpcom/v2',
 				},
-				{ dev: query.dev },
+				{ query },
 				{
 					view: ( data as any ).view_name,
+					card,
 					...( reminder && { reminder } ),
 				}
 			);
 		},
 		{
+			onMutate( { card } ) {
+				const cachedData: Record< string, unknown > | undefined = queryClient.getQueryData(
+					getCacheKey( siteId )
+				);
+				const optimisticUpdate = {
+					...cachedData,
+					primary: cachedData.primary.filter( ( c ) => c !== card ),
+				};
+
+				queryClient.setQueryData( getCacheKey( siteId ), optimisticUpdate );
+			},
+			onError(/* err, _data, cachedData */) {
```

Comment:
need to decide here too, I don't think we have to revert, because it's just for showing promo cards